extras/indexing/src/test/java/org/apache/rya/indexing/accumulo/customfunction/AccumuloCustomSparqlFunctionTest.java

```diff
@@ -0,0 +1,96 @@
+package org.apache.rya.indexing.accumulo.customfunction;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Set;
+
+import org.apache.rya.accumulo.AccumuloRdfConfiguration;
+import org.apache.rya.indexing.accumulo.ConfigUtils;
+import org.apache.rya.indexing.accumulo.freetext.SimpleTokenizer;
+import org.apache.rya.indexing.accumulo.freetext.Tokenizer;
+import org.apache.rya.sail.config.RyaSailFactory;
+import org.junit.Before;
+import org.junit.Test;
+import org.openrdf.model.Value;
+import org.openrdf.query.BindingSet;
+import org.openrdf.query.MalformedQueryException;
+import org.openrdf.query.QueryLanguage;
+import org.openrdf.query.TupleQuery;
+import org.openrdf.query.TupleQueryResult;
+import org.openrdf.repository.Repository;
+import org.openrdf.repository.sail.SailRepository;
+import org.openrdf.sail.Sail;
+
+import org.junit.*;
+
+import org.openrdf.repository.RepositoryConnection;
+import org.openrdf.repository.RepositoryException;
+
+public class AccumuloCustomSparqlFunctionTest {
+	private AccumuloRdfConfiguration conf;
+	private Repository repository = null;
+	private Sail sail = null;
+	private RepositoryConnection conn = null;
+	
+
+	@Before
+    public void before() throws Exception {
+        conf = new AccumuloRdfConfiguration();
+        conf.setBoolean(ConfigUtils.USE_MOCK_INSTANCE, true);
+        conf.set(ConfigUtils.CLOUDBASE_USER, "USERNAME");
+        conf.set(ConfigUtils.CLOUDBASE_PASSWORD, "PASS");
+        conf.set(ConfigUtils.CLOUDBASE_AUTHS, "U");
+        conf.setClass(ConfigUtils.TOKENIZER_CLASS, SimpleTokenizer.class, Tokenizer.class);
+        conf.setTablePrefix("triplestore_");
+        
+        this.sail = RyaSailFactory.getInstance(conf);
+        this.repository = new SailRepository(this.sail);
+        this.conn = this.repository.getConnection();
+        
+    }
+	
+	@Test
+	public void customFunction() {
+		String sql = "" 
+				+ "PREFIX fn: <http://example.org#> "
+				+ "select (fn:mycustomfucnction() as ?res1) (str(fn:mycustomfucnction('Rya')) as ?res2) "
+				+ "where {}";
+		try {
+			TupleQuery query = this.conn.prepareTupleQuery(QueryLanguage.SPARQL, sql);
+		
+			TupleQueryResult qres = query.evaluate();
+			ArrayList<HashMap<String, Value>> reslist = new ArrayList<HashMap<String, Value>>();
+			while (qres.hasNext()) {
+				BindingSet b = qres.next();
+				Set<String> names = b.getBindingNames();
+				HashMap<String, Value> hm = new HashMap<String, Value>();
+				for (Object n : names) {
```

Comment:
Why loop over objects when the Set's generic type is already knows to be String?